### DISCUSSION TITLE
Does this still use Nitter?

### DISCUSSION THREAD

**Discussion by candiesdoodle** [jonjomckay/fritter#76] [Q&A] [answered]
Hi, great work and thank you
Wanted to confirm whether this project still uses a public nitter instance to pull the tweets?
Saw reference to Twitter API token, hence confused.
If it's moved to directly pulling feeds from Twitter using API, can we get there option to supply own api tokens? (from Twitter developer console?

**Comment by jonjomckay** [comment, marked as answer]
No, as of v2.0.0 the app no longer uses Nitter as its source - it now uses the same hidden Twitter APIs that Nitter itself uses. As far as I'm aware, those API endpoints don't support the consumer key and secret authentication from the Developer Console, and I'm not sure it would be a great fit for Fritter. Fritter intends to be a privacy-first frontend for Twitter, and using user-specific API keys would go against that principle.

From the v2.0.0 release notes:

> **Removal of Nitter Support**
>
> Support for Nitter instances has now been removed, and is now replaced with direct support for the Twitter API, using the same methods Nitter uses under the hood.
>
> This was done mainly so that the app doesn't DOS (abuse) public Nitter instances, and also due to development issues caused by differing data structures from all the different Nitter versions being hosted in the wild.

You may be interested in [Twidere X](https://github.com/TwidereProject/TwidereX-Android), which does allow you to supply your own API keys, but be aware that it does mean Twitter would be able to track you more than using Fritter, and you would need a Twitter account to use features like subscriptions/follows 🙂
